src/pudl/metadata/fields.py

```diff
@@ -2120,6 +2152,17 @@
 
 FIELD_METADATA_BY_RESOURCE: dict[str, dict[str, Any]] = {
     "sector_consolidated_eia": {"code": {"type": "integer"}},
+    "plant_parts_eia": {
```

Comment:
This isn't the nicest to make all the `category` columns part of the resource specific metadata but when I included categorical columns in the general `FIELD_METADATA` there were some errors from `string` columns becoming `category` so I decided it's safest to do all the categorical dtype conversions at the end on a resource specific level.

Reply:
It's probably more correct for a bunch of our coded columns to be categoricals in the database, but they're almost all strings right now, and I'm sure we rely on that typing in a ton of places, so this seems fine for now. I made an issue to remember this in, if you have thoughts to add: #1866